shared/protos/Room.proto

```diff
@@ -107,6 +107,18 @@ message Room {
 
   repeated Tile tiles = 18 [(gmx) = "tiles/tile"];
 
+  message Layer {
+    optional string name = 1;
+    optional int32 id = 2;
```

Comment:
I just discovered a small problem thanks to Salvak testing. Layers are actually a global resource like rooms themselves so you can modify them outside the room. The ids are global just like instances and tiles were, so the first rooms first default layer has an id of 0 and the second room has an id of 2.